flocker/node/agents/cinder.py

```diff
@@ -380,6 +380,31 @@ def _extract_nova_server_addresses(addresses):
     return all_addresses
 
 
+def _get_compute_id(local_ips, id_to_node_ips):
+    """
+    Compute the instance ID of the local machine.
+
+    Expectation is that our local IPs intersect with one (only) of the
+    remote nodes' sets of IPs.
+
+    :param set local_ips: The local machine's IPs.
+    :param id_to_node_ips: Mapping from instance IDs to sets of IPs, as
```

Comment:
Yep, it includes IPv6. And I think that it might be Floating IPv6 addresses that are causing the problem on Ubuntu 14.04.
On other Ubuntu's the IPv6 addresses are assigned to interfaces in the OS.